src/core/ext/lb_policy/grpclb/grpclb.c

```diff
@@ -158,6 +158,12 @@ typedef struct wrapped_rr_closure_arg {
   /* when not NULL, represents a pending_{pick,ping} node to be freed upon
    * closure execution */
   void *owning_pending_node; /* to be freed if not NULL */
+
+  /* Pointer ot heap memory if the closure and its argument were allocated
+   * dynamically outside of a pending pick. It'll be NULL otherwise.
+   *
+   * TODO(dgq): This is by no means pretty. */
+  void *closure_mem_or_null;
```

Comment:
Here's an alternative suggestion that might be cleaner: Store the closure in wrapped_rr_closure_arg.  That way, we never have to free it at all, because it will be freed when wrapped_rr_closure_arg is.  This also avoids the duplication in the existing code where we currently store a closure in each of pending_pick and pending_ping.